text/builders/0000-jammy-builders.md

```diff
@@ -0,0 +1,150 @@
+# Builders based on Ubuntu 2022.04: Jammy Jellyfish
+
+## Summary
+
+A set of stacks based on the Ubuntu 2022.04 LTS (Jammy Jellyfish) will be
+maintained by Paketo, per stacks [RFC
+0004](https://github.com/paketo-buildpacks/rfcs/blob/da3339d071ffed23c3cd1b374a6bfcefdea7ac70/text/stacks/0004-jammy-jellyfish.md).
+Builders are an important way for Paketo users to consume stacks; they can be
+used as an input to the `pack` CLI. Paketo should maintain a set of builders
+that use the Jammy Jellyfish stacks.  They should be developed, released, and
+maintained by the Builders subteam.  These new builders should come in "full",
+"buildpackless-full", "base", "buildpackless-base",  "tiny", and
+"buildpackless-tiny" variants.
+
+## Motivation
+
+Paketo [plans to support Jammy Jellyfish
+stacks](https://github.com/paketo-buildpacks/rfcs/blob/da3339d071ffed23c3cd1b374a6bfcefdea7ac70/text/stacks/0004-jammy-jellyfish.md).
+Paketo [builders](https://paketo.io/docs/concepts/builders/) are a common way
+for Paketo users to consume Paketo stacks, buildpacks, and the Cloud Native
+Buildpacks (CNB) lifecycle. In order for many Paketo users to get value from
+Jammy Jellyfish stacks, we'll need to provide Jammy Jellyfish builders.
+
+## Detailed Explanation
+
+### Variants
+
+The Jammy builders will be available in 6 variants that align to the existing
```

Comment:
I don't think it makes much difference - having 12 repositories with one thing vs one repository with 12 things feels about the same to me. I could be under-estimating something, though.

In general it is easier and simpler to continue with a current pattern so I would want to see a strong reason for changing it up. I'm open to change if we can articulate clear benefits, though.

Reply:
From my experience implementing automation for the full builder + buildpackless full builder in the same repository, I found that combining multiple released artifacts in the same repository is quite painful. Ultimately, we moved away from housing both variants in the same repo because of the added complexity. Right now, builder repos represent a tiny fraction of my weekly maintenance tasks. They are mostly set and forget!

Reply:
> If we support another stack, we'll have 18 repositories, which I could imagine being a headache when some automated PR like dependabot fails and we have to go to 18 repositories and make the same change to get the PR in.

In my experience, it is easier to make a larger number of changes to lots of similar looking things (e.g. repositories with the same structure) than a smaller number of changes to things that don't look the same (e.g. repositories with different structure).

In the example above, if I had to fix 18 repositories, I would consider scripting a solution. It would apply to all repositories in the same way, so it would be cognitively simple.

A concrete example of this approach is when I removed the `ioutil` package from all of our repositories. The changes were structurally the same to all repositories, so once I found a script that worked for one or two repositories I had confidence it would apply correctly to all other repositories.

My main concern with more repositories is the additional automation burden on github actions which are already somewhat flaky for us. However, I don't think fewer repositories actually reduces the number of actions that run, and it increases complexity for the reasons stated above. I see this burden as a direct consequence of increasing our supported surface area (i.e. adding more stacks and builders), and I don't see a way to mitigate it.